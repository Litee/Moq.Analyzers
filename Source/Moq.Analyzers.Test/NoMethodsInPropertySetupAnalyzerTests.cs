using ApprovalTests;
using ApprovalTests.Reporters;
using Microsoft.CodeAnalysis.Diagnostics;
using System.IO;
using TestHelper;
using Xunit;

namespace Moq.Analyzers.Test
{
    [UseReporter(typeof(DiffReporter))]
    public class NoMethodsInPropertySetupAnalyzerTests : DiagnosticVerifier
    {

        [Fact]
        public void Test()
        {
            Approvals.Verify(VerifyCSharpDiagnostic(File.ReadAllText("Data/NoMethodsInPropertySetup.cs")));
        }

        protected override DiagnosticAnalyzer GetCSharpDiagnosticAnalyzer()
        {
            return new NoMethodsInPropertySetupAnalyzer();
        }
    }
}