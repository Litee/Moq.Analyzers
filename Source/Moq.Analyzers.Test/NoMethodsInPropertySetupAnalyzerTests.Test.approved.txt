Diagnostic 1
	Id: Moq1101
	Location: SourceFile(Test0.cs[355..365))
	Highlight: x.Method()
	Lines: mock.SetupGet(x => x.Method());
	Severity: Warning
	Message: Do not use SetupGet/SetupSet for methods.

Diagnostic 2
	Id: Moq1101
	Location: SourceFile(Test0.cs[400..410))
	Highlight: x.Method()
	Lines: mock.SetupSet(x => x.Method());
	Severity: Warning
	Message: Do not use SetupGet/SetupSet for methods.

